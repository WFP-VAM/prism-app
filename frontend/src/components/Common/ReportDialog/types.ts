import { Style } from '@react-pdf/types';
import { TFunctionResult } from 'i18next';

export interface PDFLegendDefinition {
  value: string | number | TFunctionResult;
  style: Style | Style[];
}

// This numbers depends on table header and row height
export const MAX_TABLE_ROWS_PER_PAGE = 29;
export const FIRST_PAGE_TABLE_ROWS = 9;
