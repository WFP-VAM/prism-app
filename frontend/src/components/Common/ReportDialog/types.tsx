export enum ReportType {
  Storm,
  Flood,
}
