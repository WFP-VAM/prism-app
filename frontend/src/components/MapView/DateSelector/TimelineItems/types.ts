import {
  CycloneName,
  ShortDate,
  TimeAndState,
} from 'context/anticipatoryAction/AAStormStateSlice/types';

export type DateItemStyle = {
  class: string;
  color: string;
  layerDirectionClass?: string;
  emphasis?: string;
};

export type DateJSON = Record<ShortDate, Record<CycloneName, TimeAndState[]>>;

export type WindStateReport = {
  states: TimeAndState[];
  cycloneName: string | null;
};
