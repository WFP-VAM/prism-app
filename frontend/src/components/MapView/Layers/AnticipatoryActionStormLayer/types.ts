// Q - still needed or needs to be improved?
import { FeatureCollection, Point, Feature } from 'geojson';

export enum FeaturePropertyDataType {
  analysis = 'analysis',
  forecast = 'forecast',
}

interface FeatureProperty {
  data_type: FeaturePropertyDataType;
  time: string;
  development: string;
  // maximum_wind_speed: number;
  // maximum_wind_gust: number;
  // wind_buffer_48: any;
  // wind_buffer_64: any;
}

export interface TimeSeries extends FeatureCollection<Point, FeatureProperty> {}
export interface TimeSeriesFeature extends Feature<Point, FeatureProperty> {}
