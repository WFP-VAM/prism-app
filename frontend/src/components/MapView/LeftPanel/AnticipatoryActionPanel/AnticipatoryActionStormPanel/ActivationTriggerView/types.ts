import { AACategory } from 'context/anticipatoryAction/AAStormStateSlice/parsedStormDataTypes';

export const AAPanelCategories: AACategory[] = [
  AACategory.Severe,
  AACategory.Moderate,
];

export const AADisplayCategory: {
  [key in AACategory]?: string;
} = {
  [AACategory.Severe]: ' >118 KM/H',
  [AACategory.Moderate]: ' >89 KM/H',
};
