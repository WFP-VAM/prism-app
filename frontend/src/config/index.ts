import { has, get } from 'lodash';
import { PublicClientApplication } from '@azure/msal-browser';
import afghanistan from './afghanistan';
import cambodia from './cambodia';
import cameroon from './cameroon';
import colombia from './colombia';
import cuba from './cuba';
import ecuador from './ecuador';
import global from './global';
import indonesia from './indonesia';
import jordan from './jordan';
import kyrgyzstan from './kyrgyzstan';
import mongolia from './mongolia';
import mozambique from './mozambique';
import myanmar from './myanmar';
import namibia from './namibia';
import nigeria from './nigeria';
import rbd from './rbd';
import sierraleone from './sierraleone';
import southsudan from './southsudan';
import srilanka from './srilanka';
import tajikistan from './tajikistan';
import tanzania from './tanzania';
import ukraine from './ukraine';
import zimbabwe from './zimbabwe';

// Upload the boundary URL to S3 to enable the use of the API in a local environment.
const DEFAULT_BOUNDARIES_FOLDER =
  'https://prism-admin-boundaries.s3.us-east-2.amazonaws.com';

const configMap = {
  afghanistan,
  cambodia,
  cameroon,
  colombia,
  cuba,
  ecuador,
  global,
  indonesia,
  jordan,
  kyrgyzstan,
  mongolia,
  mozambique,
  myanmar,
  namibia,
  nigeria,
  rbd,
  sierraleone,
  southsudan,
  srilanka,
  tajikistan,
  tanzania,
  ukraine,
  zimbabwe,
} as const;

type Country = keyof typeof configMap;

const DEFAULT: Country = 'mozambique';

const {
  REACT_APP_COUNTRY: COUNTRY,
  REACT_APP_OAUTH_CLIENT_ID: CLIENT_ID,
  REACT_APP_OAUTH_AUTHORITY: AUTHORITY,
  REACT_APP_OAUTH_REDIRECT_URI: REDIRECT_URI,
  REACT_APP_TESTING: TESTING,
} = process.env;

const safeCountry =
  COUNTRY && has(configMap, COUNTRY.toLocaleLowerCase())
    ? (COUNTRY.toLocaleLowerCase() as Country)
    : DEFAULT;

const {
  appConfig,
  defaultBoundariesFile,
  rawLayers,
  rawTables,
  rawReports,
}: {
  appConfig: Record<string, any>;
  defaultBoundariesFile: string;
  rawLayers: Record<string, any>;
  rawTables: Record<string, any>;
  rawReports: Record<string, any>;
} = configMap[safeCountry];

const translation = get(configMap[safeCountry], 'translation', {});

const msalConfig = {
  auth: {
    clientId: CLIENT_ID!,
    authority: AUTHORITY!,
    redirectUri: REDIRECT_URI!,
  },
};

const msalRequest = {
  scopes: ['openid', 'profile'],
};

const msalInstance = new PublicClientApplication(msalConfig);

const authRequired: boolean =
  !TESTING && get(appConfig, 'WFPAuthRequired', false);

const enableNavigationDropdown: boolean = get(
  appConfig,
  'enableNavigationDropdown',
  false,
);

const defaultBoundariesPath = `${DEFAULT_BOUNDARIES_FOLDER}/${defaultBoundariesFile}`;

export {
  appConfig,
  authRequired,
  safeCountry,
  defaultBoundariesPath,
  rawLayers,
  rawTables,
  rawReports,
  msalInstance,
  msalRequest,
  enableNavigationDropdown,
  translation,
};
