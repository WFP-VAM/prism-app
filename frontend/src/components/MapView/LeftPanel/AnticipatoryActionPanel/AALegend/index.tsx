import { ListItem, Paper, createStyles, makeStyles } from '@material-ui/core';
import { lightGrey } from 'muiTheme';
import { useSelector } from 'react-redux';
import { leftPanelTabValueSelector } from 'context/leftPanelStateSlice';
import { Panel } from 'config/types';
import AADroughtLegend from '../AADroughtLegend';
import AAStormLegend from '../AAStormLegend';

export interface AALegendProps {
  forPrinting?: boolean;
  showDescription?: boolean;
}

function AALegend({
  forPrinting = false,
  showDescription = true,
}: AALegendProps) {
  const classes = useStyles();

  const tabPanel = useSelector(leftPanelTabValueSelector);
  const isStormAA = tabPanel === Panel.AnticipatoryActionStorm;
  const isDroughtAA = tabPanel === Panel.AnticipatoryActionDrought;

  return (
    <ListItem disableGutters dense>
      <Paper
        className={classes.paper}
        elevation={forPrinting ? 0 : undefined}
        style={
          forPrinting
            ? {
                border: `1px solid ${lightGrey}`,
              }
            : undefined
        }
      >
        {isDroughtAA && <AADroughtLegend showDescription={showDescription} />}
        {isStormAA && <AAStormLegend />}
      </Paper>
    </ListItem>
  );
}

const useStyles = makeStyles(() =>
  createStyles({
    paper: {
      padding: 8,
      width: 180,
      borderRadius: '8px',
      display: 'flex',
      flexDirection: 'column',
      gap: '0.5rem',
    },
  }),
);

export default AALegend;
