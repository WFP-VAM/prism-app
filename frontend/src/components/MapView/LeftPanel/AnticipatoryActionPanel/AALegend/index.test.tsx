import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Panel } from 'config/types';
import { Provider } from 'react-redux';
import AALegend from '.';

test('renders as expected', () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    leftPanelState: {
      tabValue: Panel.AnticipatoryActionDrought,
    },
  });

  const { container } = render(
    <Provider store={store}>
      <AALegend />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
