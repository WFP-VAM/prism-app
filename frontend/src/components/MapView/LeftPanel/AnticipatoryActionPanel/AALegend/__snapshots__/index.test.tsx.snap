// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-listitem
    dense="true"
    disablegutters="true"
  >
    <div
      class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <mock-dialog
        open="false"
      >
        <mock-dialogtitle>
          <div
            class="makeStyles-titleWrapper-13"
          >
            <mock-typography
              variant="h2"
            >
              How to read this screen
            </mock-typography>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="padding: 0px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </mock-dialogtitle>
        <mock-dialogcontent
          dividers="true"
        >
          <div
            class="makeStyles-contentWrapper-14"
          >
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Window 1
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                Start to mid of the rainfall season.
              </mock-typography>
               
            </div>
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Window 2
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                Mid to end of the rainfall season.
              </mock-typography>
               
            </div>
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Mild category
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                Drought events that typically occur once every 4 years.
              </mock-typography>
               
            </div>
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Moderate category
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                Drought events that typically occur once every 5 years.
              </mock-typography>
               
            </div>
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Severe category
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                Drought events that typically occur once every 7 years.
              </mock-typography>
               
            </div>
            <div
              class="makeStyles-contentItem-15"
            >
              <mock-typography
                style="font-weight: bold;"
                variant="h3"
              >
                Ready, Set and Go phases
              </mock-typography>
               
              <mock-typography
                variant="h3"
              >
                The "Ready, Set & Go!" system uses seasonal forecasts with longer lead time for preparedness (Ready phase) and shorter lead times for activation and mobilization (Set & Go! phases).
              </mock-typography>
               
            </div>
          </div>
        </mock-dialogcontent>
        <mock-dialogactions
          classname="makeStyles-dialogActionsWrapper-16"
        >
          <mock-button
            classname="makeStyles-dialogButton-17"
            component="a"
            href="mailto:anticipatory_action@wfp.org"
            starticon="[object Object]"
            target="_blank"
            variant="outlined"
          >
            <mock-typography>
              I STILL NEED HELP
            </mock-typography>
          </mock-button>
          <mock-button
            classname="makeStyles-dialogButton-17"
            starticon="[object Object]"
            variant="outlined"
          >
            <mock-typography>
              Close
            </mock-typography>
          </mock-button>
        </mock-dialogactions>
      </mock-dialog>
      <mock-typography
        style="font-weight: bold;"
        variant="h3"
      >
        Phases
      </mock-typography>
      <div
        style="display: flex; flex-direction: column; gap: 0.5rem; margin-bottom: 0.75rem;"
      >
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid white; color: white;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid white; background: rgb(131, 31, 0);"
              >
                S
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(131, 31, 0);"
              >
                SEV
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Set
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Severe
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid white; color: white;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid white; background: rgb(230, 55, 1);"
              >
                R
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(230, 55, 1);"
              >
                SEV
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Ready
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Severe
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid black; background: rgb(255, 137, 52);"
              >
                S
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(255, 137, 52);"
              >
                MOD
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Set
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Moderate
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid black; background: rgb(255, 213, 45);"
              >
                R
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(255, 213, 45);"
              >
                MOD
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Ready
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Moderate
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid black; background: rgb(255, 245, 3);"
              >
                S
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(255, 245, 3);"
              >
                MIL
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Set
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Mild
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="border-bottom: 1px solid black; background: rgb(255, 252, 179);"
              >
                R
              </div>
              <div
                class="makeStyles-bottomTextContainer-21"
                style="background: rgb(255, 252, 179);"
              >
                MIL
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Ready
            </mock-typography>
            <mock-typography
              style="white-space: nowrap;"
            >
              Mild
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="background: rgb(241, 241, 241);"
              >
                na
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              No Action
            </mock-typography>
          </div>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-iconWrapper-18"
          >
            <div
              class="makeStyles-centerContainer-19"
              style="border: 1px solid black; color: black;"
            >
              <div
                class="makeStyles-topTextContainer-20"
                style="background: rgb(241, 241, 241);"
              >
                ny
              </div>
            </div>
          </div>
          <div>
            <mock-typography
              style="white-space: nowrap;"
            >
              Not Yet Monitored
            </mock-typography>
          </div>
        </div>
      </div>
      <mock-typography>
        <span
          class="makeStyles-dialogButton-3"
          role="button"
          tabindex="0"
        >
          The "Ready, Set & Go!" system
        </span>
         
        uses seasonal forecasts with longer lead time for preparedness (Ready phase) and shorter lead times for activation and mobilization (Set & Go! phases).
      </mock-typography>
      <mock-divider />
      <mock-typography
        style="font-weight: bold;"
        variant="h3"
      >
        Districts
      </mock-typography>
      <div
        style="display: flex; flex-direction: column; gap: 0.5rem;"
      >
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            style="min-width: 2.2rem; border: 1px solid #a4a4a4; border-radius: 2px;"
          />
          <mock-typography>
            District
          </mock-typography>
        </div>
        <div
          class="makeStyles-itemWrapper-2"
        >
          <div
            class="makeStyles-newTag-11"
          >
            NEW
          </div>
          <mock-typography>
            District in new phase this month
          </mock-typography>
        </div>
      </div>
    </div>
  </mock-listitem>
</div>
`;
